lib/Parse/ParseExpr.cpp

```diff
@@ -2923,6 +2923,12 @@ Expr *Parser::parseExprAnonClosureArg() {
   auto closure = dyn_cast_or_null<ClosureExpr>(
       dyn_cast<AbstractClosureExpr>(CurDeclContext));
   if (!closure) {
+    if (Context.LangOpts.DebuggerSupport) {
+      auto refKind = DeclRefKind::Ordinary;
```

Comment:
Can you include a comment as to why this is necessary?

Reply:
Sure, what about: 
```
lldb uses `$` to denote expression parser results and persistent variables so instead of emitting a diagnostic and failing we return an unresolved reference expression for the debugger to fill in
```

?

Reply:
That's something we already handle generally with dollar identifiers, though. I was thinking something more specific to closures:

> It's possible we're at a breakpoint that's stopped in the middle of a closure with anonymous arguments. In that case, we want to give LLDB a chance to resolve the identifier.

Reply:
OK, sure!